modules/standard/Data/Promise.enc

```diff
@@ -0,0 +1,32 @@
+-- Promises are futures that you can fulfil by explictly calling a method that you pass a value.
+module Promise
+
+-- Create a promise, together with a future that can be
+-- fulfilled using the promise:
+-- Once you call fulfil(..) on the promise object, the
+-- future returned by this call will be fulfilled.
+fun mkPromise[t]() : (Fut[t], Promise[t])
+  val f = EMBED (Fut[t]) future_mk(_ctx, _enc__type_t); END
+  (f, new Promise(f))
+end
+
+-- This should be linear, but we're hitting issue #770 then
+-- in the mean time, do not alias promises!
+unsafe class Promise[t]
```

Comment:
@TobiasWrigstad What exactly do you mean `that`? Is that referring to the trait-based decomposition. There was no linearity then.
Also, I don't think my proposal is good enough. You want to consume the promise after fulfilling that. Probably needs a function.